drivers/sensor/bmi160/bmi160.h

```diff
@@ -398,8 +398,29 @@ struct bmi160_range {
 	uint8_t reg_val;
 };
 
+#define BMI160_BUS_SPI		DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+
+union bmi160_bus_cfg {
```

Comment:
This is personal preference, but can you nest this union down a level, please?

```
struct bmi_bus_cfg {
   union {
#if BM160_BUS_SPI
	const struct spi_config *spi_cfg;
#endif
  }
};
```

It will take the same about of space and it will give you a place to put common config if the need arise (and you don't have to pass around a dirty union in your function signatures)

Reply:
done